packages/salesforcedx-vscode-core/src/conflict/messages.ts

```diff
@@ -0,0 +1,45 @@
+import { SfdxCommandBuilder } from '@salesforce/salesforcedx-utils-vscode';
+import { ConflictDetectionMessages } from '../commands/util';
+
+const messagesByLogName: Map<string, ConflictDetectionMessages> = new Map();
```

Comment:
Nit-pick here.  `messagesByLogName` and `warningMessageKey` get created when this file is imported.  That in of itself isn't too bad, but then `messagesByLogName.set()` is called twice.  So when this file is imported, that means this file executes code, and this makes it hard to write unit tests for.

I would rather see something like this:
```
import { SfdxCommandBuilder } from '@salesforce/salesforcedx-utils-vscode';
import { ConflictDetectionMessages } from '../commands/util';

export function getConflictMessagesFor(
  logName: string
): ConflictDetectionMessages | undefined {
  const messagesByLogName: Map<string, ConflictDetectionMessages> = new Map();
  const warningMessageKey = 'conflict_detect_conflicts_during_deploy';

  messagesByLogName.set('force_source_deploy_with_sourcepath_beta', {
    warningMessageKey,
    commandHint: inputs => {
      const commands: string[] = [];
      (inputs as string[]).forEach(input => {
        commands.push(
          new SfdxCommandBuilder()
            .withArg('force:source:deploy')
            .withFlag('--sourcepath', input)
            .build()
            .toString()
        );
      });
      const hints = commands.join('\n  ');

      return hints;
    }
  });

  messagesByLogName.set('force_source_deploy_with_manifest_beta', {
    warningMessageKey,
    commandHint: input => {
      return new SfdxCommandBuilder()
        .withArg('force:source:deploy')
        .withFlag('--manifest', input as string)
        .build()
        .toString();
    }
  });

  const conflictMessages = messagesByLogName.get(logName);
  if (!conflictMessages) {
    throw new Error(`No conflict messages found for ${logName}`);
  }
  return conflictMessages;
}
```
If I'm missing something here, let's discuss f2f.